src/icu.rs

```diff
@@ -54,11 +54,15 @@ pub fn set_common_data_71(data: &'static [u8]) -> Result<(), i32> {
 
 /// Returns BCP47 language tag.
 pub fn get_language_tag() -> String {
-  let mut output = [0u8; 1024];
-  let len = unsafe {
-    icu_get_default_locale(output.as_mut_ptr() as *mut c_char, output.len())
-  };
-  std::str::from_utf8(&output[..len]).unwrap().to_owned()
+  let mut output = Vec::<u8>::with_capacity(1024);
+  unsafe {
+    let len = icu_get_default_locale(
+      output.as_mut_ptr().cast::<c_char>(),
```

Comment:
This is just trading one UB to another: You'll need to use Vec's `spare_capacity_mut` method to get a slice of `MaybeUninit`s.

Reply:
See the other review comments

Reply:
Mmm, I would agree with your response that the insides of `spare_capacity_mut` is an implementation detail, and I'd argue that the example in `as_mut_ptr`'s documentation is probably written before `MaybeUninit<T>` was even a thing, much less `spare_capacity_mut`. But it's your call.